### PR TITLE
feature/codacy-and-documentation-update

### PR DESCRIPTION
**Pull request by mauriciospinardi-cloudwalk**
This PR will:

```markdown
## [1.1.7] - YYYY-MM-DD
- Replace `BuildConfig#PACKAGE_ID` by `BuildConfig#LIBRARY_PACKAGE_NAME`
- Update documentation.
- Remove previously deprecated `Application#getPackageContext()`.
- Update target SDK version to 33.
- Update code patterns under Codacy community guidelines.
- Update `Log#getByteTrace(String, int, int)` and `Log#h(String, int, int)`
  argument order, swapping `length` and `offset` to comply with community
  standards.
- Review CHANGELOG.md content.
```

### CODE REVIEW

**Comment by mauriciospinardi-cloudwalk** [comment]
@cw-romedeiros @cw-luciano-santos take a look and let me know if you see any blocker.
(I'm currently setting up this repo with the administrators to properly have you as reviewers)

**Comment by mauriciospinardi-cloudwalk** [comment]
By the way, there are no functional tests to be made at this moment. Compilation and publication are the main targets of review.

File: LogLibrary/src/main/java/io/cloudwalk/loglibrary/internals/utilities/SnifferUtility.java [hunk 1]
```diff
@@ -103,7 +110,7 @@ public void run() {
 
                         sPath = root + "/" + name + ".log";
 
-                        if (!(new File(sPath)).exists()) { _clear(); continue; }
+                        if (!(new File(sPath)).exists()) { clear(); continue; }
```
Comment (cw-luciano-santos) [inline LogLibrary/src/main/java/io/cloudwalk/loglibrary/internals/utilities/SnifferUtility.java:113]: This inline if statement should be broken in diferent lines

File: LogLibrary/src/main/java/io/cloudwalk/loglibrary/internals/utilities/SnifferUtility.java [hunk 2]
```diff
@@ -124,23 +131,22 @@ public void run() {
...
+            StringWriter sw = new StringWriter();
+            PrintWriter  pw = new PrintWriter(sw);
 
-            sPath = null;
+            tr.printStackTrace(pw);
 
-            list = Application.getContext().getExternalFilesDir("Log").listFiles();
-        } catch (Exception exception) {
-            android.util.Log.e(TAG, android.util.Log.getStackTraceString(exception));
+            stack = sw.toString();
```
Comment (cw-luciano-santos) [inline LogLibrary/src/main/java/io/cloudwalk/loglibrary/internals/utilities/SnifferUtility.java:143]: To understand what "sw" means you have to remember of its instantiation or the reader will search again for it.
"Use words that can be pronounced and avoid abbreviations" (Clean Code - Uncle Bob)
Same for "tr", "msg", "pw"

File: LogLibrary/src/main/java/io/cloudwalk/loglibrary/internals/utilities/SnifferUtility.java (hunk 1 above)
Comment (mauriciospinardi-cloudwalk) [reply LogLibrary/src/main/java/io/cloudwalk/loglibrary/internals/utilities/SnifferUtility.java]: Fixed with commit [7becbd8](https://github.com/cloudwalk/android-misc-loglibrary/pull/1/commits/7becbd8ada229fb8e6458a8ddc12156ee903574e).

File: LogLibrary/src/main/java/io/cloudwalk/loglibrary/internals/utilities/SnifferUtility.java (hunk 2 above)
Comment (mauriciospinardi-cloudwalk) [reply LogLibrary/src/main/java/io/cloudwalk/loglibrary/internals/utilities/SnifferUtility.java]: Fixed with commit [7becbd8](https://github.com/cloudwalk/android-misc-loglibrary/pull/1/commits/7becbd8ada229fb8e6458a8ddc12156ee903574e) for `sw`.
`tr` and `msg`, however, are arguments mirroring the signature of the default `android.util.Log` methods, so they will be kept as is, for now.

**Review by cw-luciano-santos — APPROVED** [review]

**Pull request merged**